content/docs/buildpacks/configuration.md

```diff
@@ -0,0 +1,258 @@
+
+Paketo Buildpacks will look for bindings in the `/platform/bindings` directory at build-time and in `$SERVICE_BINDING_ROOT` or `$CNB_BINDINGS` directory at runtime.
+
+For example, the Java Buildpack accepts a binding with `type` equal to `maven` containing a key named `settings.xml` containing [Maven settings][maven settings]. In the build container, the Maven Buildpack will use `settings.xml` if it finds either
+{{< code/copyable >}}
+/platform
+└── bindings
+    └── <name>
+        ├── settings.xml
+        └── type
+{{< /code/copyable >}}
+or
+{{< code/copyable >}}
+/platform
+└── bindings
+    └──<name>
+        └── metadata
+        |   └── kind
+        └── secret
+            └── settings.xml
+{{< /code/copyable >}}
+on the filesystem, where either the `type` or `kind` file contains the string `maven`.
+
+
+#### How to use bindings?
+The workflow for creating a binding and providing it to a build will depend on the chosen platform. For example, `pack` users should use the `--volume` flag to mount a binding directory into the build or app containers. Users of the `kpack` platform should store key value pairs in a Kubernetes Secret and provide that secret and associated metadata to an Image as described in the [kpack documentation][kpack service bindings].
+
+
+**Example:**  Providing a Binding to `pack build`
+
+Given a directory containing a build-time binding, `pack` users can provide this binding to a Paketo buildpack using the `--volume` flag.
+{{< code/copyable >}}
+pack build --volume <absolute-path-to-binding>:/platform/bindings/<binding-name> <image-name>
+{{< /code/copyable >}}
+
+**Example:**  Providing a Binding to `docker run`
+
+Given a directory containing a runtime binding, `docker` users can provide the binding to the app image using the `--volume` and `--env` flags
+{{< code/copyable >}}
+pack build --env SERVICE_BINDING_ROOT=/bindings --volume <absolute-path-to-binding>:/bindings/<binding-name> <image-name>
+{{< /code/copyable >}}
+
+### Procfiles
+Paketo users may override buildpack-provided types or augment the app-image with additional process types using a `Procfile`.  `Procfile` support is provided by the [Paketo Procfile Buildpack][bp/procfile]. The Procfile Buildpack will search for a file named `Procfile` at the root of the application. `Procfiles` should adhere to the following schema:
+```
+<type>: <command>
+```
+
+If a given [language family buildpack][language family buildpacks] does not contain the Procfile Buildpack it can be explicitly appended at runtime.
+
+**Example**: A Hello World Procfile
+
+The following adds a process with `type` equal to `hello` and makes it the default process.
+{{< code/copyable >}}
+echo "hello: echo hello world" > nodejs/yarn
+pack build samples/nodejs \
+  --path nodejs/yarn \
+  --buildpack paketo-buildpacks/nodejs \
+  --buildpack paketo-buildpacks/procfile \
```

Comment:
When I tried this as specified, I see the error `ERROR: failed to build: processing order: unable to resolve version: multiple versions of paketo-buildpacks/procfile - must specify an explicit version`. Should we instead just reference the GCR path for procfile?

Reply:
Yeah good catch, I am seeing the same behavior. This actually seems like a `pack` bug to me, given there is only one version of `paketo-buildpacks/procfile` on the builder image. I'll switch to using the image reference for now and file a bug with `pack`.

Reply:
https://github.com/buildpacks/pack/issues/858